sources/academy/platform/getting_started/creating_actors.md

```diff
@@ -1,131 +1,180 @@
+You will end up on a template detail page where you can see the all the important information about the template - description, included features, used technologies, and what is the use-case of this template. More importantly, there is code preview and also instructions for how the code works.
 
-```js
-const result = (input?.firstNumber || 0) + (input?.secondNumber || 0);
+![JavaScript template detail page](./images/create-actor-template-detail-page.png)
+
+### Using the template in the Web IDE {#web-ide}
+
+By clicking **Use this template** button you will create the Actor in Apify Console and you will be moved to the **Code** tab with the [Web IDE](/platform/actors/development/quick-start/web-ide) where you can see the code of the template and start editing it.
+
+> The Web IDE is a great tool for developing your Actor directly in Apify Console without the need to install or use any other software.
+
+![Web IDE](./images/create-actor-web-ide.png)
+
+### Using the template locally {#local}
+
+If you want to use the template locally, you can again use our [Apify CLI](/cli) to download the template to your local machine.
+
+> Creating an Actor from a template locally is a great option if you want to develop your code using your local environment and IDE and then push the final solution back to the Apify Console.
+
+When you click on the **Use locally** button, you'll be presented with instructions on how to create an Actor from this template in your local environment.
+
+With the Apify CLI installed, you can simply run following commands in your terminal:
+
+```bash
+apify create my-actor -t getting_started_node
+```
+
+```bash
+cd my-actor
+apify run
 ```
 
-The final code should look like this:
+![Use the template locally](./images/create-actor-template-locally.png)
+
+## Start with scraping single page {#scraping-single-page}
+
+This template is a great starting point for web scraping as it extracts data from a single website. It uses [Axios](https://axios-http.com/docs/intro) for downloading the page content and [Cheerio](https://cheerio.js.org/) for parsing the HTML from the content.
+
+Let's see what's inside the **Start with JavaScript** template. The main logic of the template lives in the `src/main.js` file.
 
 ```js
+// Axios - Promise based HTTP client for the browser and node.js (Read more at https://axios-http.com/docs/intro).
+import axios from 'axios';
+// Cheerio - The fast, flexible & elegant library for parsing and manipulating HTML and XML (Read more at https://cheerio.js.org/).
+import * as cheerio from 'cheerio';
+// Apify SDK - toolkit for building Apify Actors (Read more at https://docs.apify.com/sdk/js/).
 import { Actor } from 'apify';
 
+// The init() call configures the Actor for its environment. It's recommended to start every Actor with an init().
 await Actor.init();
 
-console.log('Loading input');
-
+// Structure of input is defined in input_schema.json
 const input = await Actor.getInput();
```

Comment:
we were discussing this some time ago, the `getInput` can return `null`, and the new code does not respect that anyhow. we were adding `getInputOrThrow` exactly for this use case, although the elders didn't like it... :] 

I'd be in favor of adopting the new method, but I don't have any energy to argue about "things not in spec", so I guess feel free to keep it the way it is now... I can understand it might feel easier to read (but that's why we added the new method), but it's not correct and may end with a `TypeError` (when executed locally without having the `INPUT.json` present). Note that the code as is would also result in TS compilation failure.

The end goal is to handle the defaults from the input schema on SDK level, but since we are not there (and it will take some time)...